modules/heat_conduction/src/kernels/HeatConductionTimeDerivative.C

```diff
@@ -17,7 +17,7 @@ InputParameters validParams<HeatConductionTimeDerivative>()
   // over current volume by setting the use_displaced_mesh flag.
   params.set<bool>("use_displaced_mesh") = true;
 
-  params.addParam<MaterialPropertyName>("specific_heat", "specific_heat",
+  params.addParam<MaterialPropertyName>("specific_heat_name", "specific_heat",
```

Comment:
I agree with @dschwen that we shouldn't have the `_name` postfix (although I recently added one myself (`eigenstrain_names`).

This isn't the only place we need some conventions on naming.  We don't have a convention on whether you put the type of the object in its name (e.g. `SideIntegralPostprocessor`).  There are lots of cases of both.  I would love to help come to some consensus on what the rules are, write them down in the Coding Standards, and then start uniformly following them.  Conventions for these kinds of things are far more important than code style conventions because this stuff breaks everyone's input files when we change our minds, while code style might be ugly, but it doesn't directly break anything.